src/to-have-style.js

```diff
@@ -0,0 +1,75 @@
+import {parse} from 'css'
+import {matcherHint} from 'jest-matcher-utils'
+import jestDiff from 'jest-diff'
+import chalk from 'chalk'
+import {checkHtmlElement} from './utils'
+
+function parseCSS(css) {
+  const ast = parse(`selector { ${css} }`, {silent: true}).stylesheet
+  if (ast.parsingErrors && ast.parsingErrors.length > 0) {
+    const {reason, line, column} = ast.parsingErrors[0]
+    return {
+      parsingError: `Syntax error parsing expected css: ${reason} in ${line}:${column}`,
+    }
+  }
+  const parsedRules = ast.rules[0].declarations
+    .filter(d => d.type === 'declaration')
+    .reduce(
+      (obj, {property, value}) => Object.assign(obj, {[property]: value}),
+      {},
+    )
+  return {parsedRules}
+}
+
+function isSubset(styles, computedStyle) {
+  return Object.entries(styles).every(
+    ([prop, value]) => computedStyle.getPropertyValue(prop) === value,
+  )
+}
+
+function printoutStyles(styles) {
```

Comment:
I thought about it, but there's a problem with this. The `css` library seems to only be able to handle the syntax of css to the stylesheet level, that is, `property: value;` pairs listed **inside** a `selector { ... }`. And here I'm handling only plain css property/value pairs (well, sets of property/value pairs really) abstracted away from a selector. I do not even think media queries apply.

For instance, I'd have to convert what `getComputedStyles` give me, which is, again, a set of plain css property/value pairs, and manually build the AST that the `css` lib would have created out of those rules with a fake css selector wrapping them, just to make `css.stringify` be able to process it.

I saw the point of using `css` to parse because even though I had to fake wrapping the plain css props in a selector, it gives me syntax error handling, which is amazing for the pretty error messages when expectations in tests fail. But after all this reasoning do you still think is worth using `css.stringify` @kentcdodds?

Reply:
(Do we even need to handle media queries here? Maybe I'm missing out some potential great use case for this, but I don't see it yet)

Reply:
You're right, I don't think we need to use `css.stringify` :+1: